geojson_modelica_translator/modelica/csv_modelica.py

```diff
@@ -27,53 +27,91 @@
+# !/usr/bin/env python
 # coding: utf-8
 
 import os
+
 import pandas as pd
 
+
 class CSVModelica(object):
 
-   def __init__(self,scaffold,res_dir, input_csv_file_name):
-        self.scaffold= scaffold
-        self.res_dir = scaffold.load_path.resources_dir
+    def __init__(self, input_csv_file_path):
+        """
+        Convert a CSV file into the format required by Modelica. This is specific to the Mass Flow Rate file
+        only as the header of the file adds in the nominal heating and cooling flow rates. Note that the columns
+        of the CSV file need to be massFlorRateHeating
+
+        :param input_csv_file_path: string, path to file to convert.
+        """
+
+        if not os.path.exists(input_csv_file_path):
+            raise Exception(f"Unable to convert CSV file because it does not exist: {input_csv_file_path}")
+
         # read the whole data set
-        timeseries_output = pd.read_csv(input_csv_file_name)
-        # round the data command
-        timeseries_output= timeseries_output.round(2)
-        # copy the first line
-        timeseries_15min= timeseries_output.loc[[0],:]
-        timeseries_output = pd.concat([timeseries_output,timeseries_15min]).sort_index()
+        self.timeseries_output = pd.read_csv(input_csv_file_path)
+        # round the data command since energyplus reports too many sigfigs.
+        self.timeseries_output = self.timeseries_output.round(2)
+        # copy the first line since Dymola wants to have time start at zero.
+        timeseries_15min = self.timeseries_output.loc[[0], :]
+        self.timeseries_output = pd.concat([self.timeseries_output, timeseries_15min]).sort_index()
         # reset index
-        timeseries_output = timeseries_output.reset_index(drop=True)
-        #Nominal massFlowRate
-        nominalHea_massFloRate=pd.DataFrame({'#heating':['#Nominal heating water mass flow rate'],'#value':    [timeseries_output['massFlowRateHeating'].max()]},
-                                                columns=['#heating','#value'])
-        nominalCoo_massFloRate=pd.DataFrame({'#cooling':['#Nominal chilled water mass flow rate'],'#value':[timeseries_output['massFlowRateCooling'].max()]},
-                                                columns=['#cooling','#value'])
-
-    def timeseries_to_modelica_data(file_path, EnergyPlustimestep, data_type, output_modelica_file_name):
-# evaluate dimensions of the matrix
-        size = timeseries_output.shape
-# modifiy the index for modelica mos
-        timeseries_output.index = (timeseries_output.index)*EnergyPlustimestep
-        timeseries_output.index.name = '#time'
-        timeseries_output.drop(['Date/Time'], axis=1, inplace=True)
-# write to csv for modelica
-        file = file_name + '.csv'
-        with open(file,'w') as f:
+        self.timeseries_output = self.timeseries_output.reset_index(drop=True)
+
+        # Note that the file that is in this test repo does not have the column you are looking for (massFlowRateHeating),
```

Comment:
The mass_flow_rates file that you sent does not have the massFlowRateheating or massFlowRateCooling columns. I think this will work once those columns exist.

Reply:
Thinking further... we should write an assertion in case the columns don't exist and let this fail nicely. Maybe work for another week.